libs/langchain/langchain/callbacks/tracers/base_async.py

```diff
@@ -0,0 +1,249 @@
+import asyncio
+from abc import ABC
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+from langchain.callbacks.base import AsyncCallbackHandler
+from langchain.callbacks.tracers.base import BaseTracer
+from langchain.callbacks.tracers.schemas import Run
+from langchain.schema.document import Document
+from langchain.schema.output import ChatGenerationChunk, GenerationChunk, LLMResult
+
+
+class InnerSyncTracer(BaseTracer):
+    def _persist_run(self, run: Run) -> None:
+        """The Langchain Tracer uses Post/Patch rather than persist."""
+
+
+class AsyncBaseTracer(AsyncCallbackHandler, ABC):
+    def __init__(self) -> None:
+        self.lock = asyncio.Lock()
+        self.inner_tracer = InnerSyncTracer()
+
+    async def on_llm_start(
+        self, serialized: Dict[str, Any], prompts: List[str], **kwargs: Any
+    ) -> None:
+        async with self.lock:
+            llm_run = self.inner_tracer.on_llm_start(serialized, prompts, **kwargs)
+
+        await asyncio.gather(
+            self._on_llm_start(llm_run),
+            self._on_run_create(llm_run),
+            return_exceptions=True,
```

Comment:
Is it useful to log a warning or an error to notify the user that there's an issue in the event an exception is enounctered during gather